caiman/source_extraction/cnmf/estimates.py

```diff
@@ -1162,6 +1162,62 @@ def manual_merge(self, components, params):
         if self.coordinates is not None:
             self.coordinates = caiman.utils.visualization.get_contours(self.A,\
                                 self.dims, thr_method='max', thr='0.2')
+            
+        self.evaluate_components(imgs, params, dview=None)
+        
+    def manual_add(self, components, imgs, imgs_residual, params):
+        ''' Manual add new neuron that is calculated from nmf.   
+        ```
+        components = [A_new, C_new]
+        ```
+
+        Args:
+            components: list 
+                components list is made of two elements. The first is the new spatial
+                component, the second is the new temporal component.
+            imgs: movie object
+                raw movie
+            imgs_residual: movie object
+                movie subtract from the signal and the background
+            params: params object
+                
+        Returns:
+            self: estimates object
+        '''
+        A_merged = components[0]
+        C_merged = components[1]
+        A_to_norm = np.sqrt(A_merged.T.dot(A_merged)).item()
+        A_merged /= A_to_norm
+        C_merged *= A_to_norm
+        
+        self.A = scipy.sparse.hstack((self.A.tocsc(), A_merged))
+        self.C = np.vstack((self.C, C_merged))
+        self.idx_components = np.arange(self.idx_components.shape[0]+1, dtype=int)
+        self.compute_residuals(np.array(imgs))
+        self.YrA = self.R
+        
+        Acsc = self.A.tocsc()[:, -1]
+        Ctmp = np.array(self.C[-1]) + np.array(self.YrA[-1])
+        computedA = Acsc
+        computedC = Ctmp
+        
+        #r = ((Acsc.T.dot(computedA)).todense() * Ctmp)/(computedA.T.dot(computedA)).todense() - computedC
+        #c_in =  np.array(computedC+r).squeeze()
+        
+        c_in = computedC
+        deconvC, bm, cm, gm, sm, ss, lam_ = constrained_foopsi(
+            c_in, g=None, **params.temporal)        
+        self.C[-1] = deconvC
+        self.YrA[-1] = c_in-deconvC        
+        self.S = np.vstack((self.S, ss))
+        self.bl = np.hstack((self.bl,np.array(bm)))
+        self.c1 = np.hstack((self.c1,np.array(cm)))
+        self.sn = np.hstack((self.sn,np.array(sm))) 
+        self.g = np.vstack((np.vstack(self.g), gm))
+        self.nr = self.nr + 1
+        
+        self.evaluate_components(imgs, params, dview=None)
```

Comment:
this looks like it will call the `evaluate_components` method on the whole set of components and not just the one.